### DISCUSSION TITLE
KiCAD 8 Bulk-edit of Symbol Fields can create empty field impacting variant computations

### DISCUSSION THREAD

**Discussion by mdeweerd** [INTI-CMNB/KiBot#611] [General]
This is for information and could also lead to a solution with regards to empty fields for variants.

I "pulled my hair" on this one.  I was suddenly getting bad groupings due to empty value fields resulting in an invalid BOM.

The empty "Value" in the BOM seems to be due to some operation in the Symbol Fields Bulk editor.  Just applying a change applied to multiple components was enought to add empty fields to all components.  Example: "kicost.vneg:Value" was added to all the components with an empty value.

The empty "Value" was set because of the empty variant values.  And I guess that other values were also unset in this case.

In some cases I want a value to become empty in certain variants.

In case "Value" needs to be empty, a "dnp" can be set for the variant, but for alternative parts (S1PN,S2PN) or distributer references (mouser#,digikey#) this is not an option.

So "empty" values are idealy ignored and when an empty value is needed they should be replacable with a specific 'keyword' such as 'empty', 'none' .
That is at least in part the task of tools that KiBOT depends on or possibly kibot could provide a uniform solution.

Anyway, if anyone stumbles on this kind of issue, checking the "empty fields" is a place to start looking.

Related KiCAD report: https://gitlab.com/kicad/code/kicad/-/issues/17831.

**Comment by set-soft** [comment]
> In some cases I want a value to become empty in certain variants.
> 
> In case "Value" needs to be empty, a "dnp" can be set for the variant, but for alternative parts (S1PN,S2PN) or distributer references (mouser#,digikey#) this is not an option.

Why do you need to have an empty *Value* field? You have various mechanisms to mark a component as DNP.

**Comment by mdeweerd** [comment]
> Why do you need to have an empty Value field?

You are correct, I think I was continuing to reuse the core issue at hand.  However, I want to empty the alternative manufacturer fields if there are none (S1PN, S1MN, S2PN, S2MN).
